src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import decimal
+import inspect
+import re
+import sys
+
+from functools import reduce
+from pathlib import Path
+from typing import Any, List, Optional
+
+import snowflake.snowpark._internal.proto.ast_pb2 as proto
+
+NoneType = type(None)
```

Comment:
nit: maybe make this variables private to the module (i.e., by prefixing with `_` and then exporting only the public interfaces?)